### PR TITLE
TRSM_EX API

### PR DESCRIPTION
**Pull request by mahmoodw**
resolves #172044

Summary of proposed changes:
-  New API to separate trtri from trsm and allow user to manage memory
-  User will need to use trtri_batched to setup invA matrix to be sent in to trtri_ex

Notes:
- Need to add explained usage of trsm_ex. Example can be found in testing_trsm_ex.
- Trtri_batched is in beta stage to support important sizes. Additional memory is required to implement n>32 when n is not a power of 2.

### CODE REVIEW

File: clients/benchmarks/client.cpp [hunk 1]
```diff
@@ -101,6 +102,8 @@ struct perf_blas<
             testing_gemm_strided_batched<T>(arg);
         else if(!strcmp(arg.function, "trsm"))
             testing_trsm<T>(arg);
+        else if(!strcmp(arg.function, "trsm_ex"))
+            testing_trsm<T>(arg);
```
Comment (leekillough) [inline clients/benchmarks/client.cpp]: This is a typo -- it should be calling `testing_trsm_ex<T>(arg);`

**Comment by leekillough** [comment]
When this is merged, please "squash and merge".

File: library/include/rocblas-functions.h [hunk 2]
```diff
@@ -1088,6 +1088,31 @@ ROCBLAS_EXPORT rocblas_status rocblas_dtrsm(rocblas_handle handle,
                                             double* B,
                                             rocblas_int ldb);
 
+/*
+ * ===========================================================================
+ *    BLAS extensions
+ * ===========================================================================
+ */
+
```
Comment (saadrahim) [inline library/include/rocblas-functions.h]: Doxygen/sphinx documentation missing for readthedocs

File: library/src/blas3/rocblas_trsm.cpp [hunk 3]
```diff
@@ -14,6 +14,21 @@
 #include "logging.h"
 #include "utility.h"
 
+template <typename T>
+void printMatrix(const char* name, T* A, rocblas_int m, rocblas_int n, rocblas_int lda)
```
Comment (saadrahim) [inline library/src/blas3/rocblas_trsm.cpp]: Is this debug code or intended to inclusion in the library?

File: library/src/blas_ex/rocblas_trsm_ex.cpp [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+                               rocblas_datatype compute_type,
+                               rocblas_trsm_option option,
+                               size_t* x_temp_size,
+                               void* x_temp_workspace)
+
+{
+    if(!x_temp_workspace)
+    {
+        if(option == rocblas_trsm_high_performance)
+            *x_temp_size = m * n;
```
Comment (saadrahim) [inline library/src/blas_ex/rocblas_trsm_ex.cpp]: what happens if x_temp_size is incorrectly specified?

**Pull request merged**